rootless-install.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+# This script is meant for quick & easy install via:
+#   $ curl -fsSL https://rootless.docker.com -o get-docker.sh
+#   $ sh get-docker.sh
+#
+# NOTE: Make sure to verify the contents of the script
+#       you downloaded matches the contents of install.sh
+#       located at https://github.com/docker/docker-install
+#       before executing.
+#
+# Git commit from https://github.com/docker/docker-install when
+# the script was uploaded (Should only be modified by upload job):
+SCRIPT_COMMIT_SHA=UNKNOWN
+
+# This script should be run with an unprivileged user and install/setup Docker under $HOME/bin/.
+
+
+# TODO:
+
+BIN="$HOME/bin"
+DAEMON=dockerd
+
+checks() {
+	# OS verification: Linux only, point osx/win to helpful locations
+	case "$(uname)" in
+	Linux)
+		;;
+	*)
+		>&2 echo "Rootless Docker cannot be installed on $(uname)"; exit 1
+		;;
+	esac
+
+	# User verification: deny running as root (unless forced?)
+	if [ "$(id -u)" = "0" ]; then
+		>&2 echo "Refusing to install rootless Docker as the root user"; exit 1
+	fi
+
+	# HOME verification
+	if [ ! -d "$HOME" ]; then
```

Comment:
and XDG_RUNTIME_DIR

Reply:
If no `XDG_RUNTIME_DIR` could we detect `/run/user/<uid>` or maybe even fall back to `/tmp/docker-rootless-<uid>`